handler.go

```diff
@@ -168,14 +167,36 @@ func (s *RPCServer) handleReader(ctx context.Context, r io.Reader, w io.Writer,
 				s.maxRequestSize))
 		return
 	}
+	// check if bufferedRequest is array or not.
+	if bufferedRequest.Bytes()[0] != '[' {
+		if err := json.Unmarshal(bufferedRequest.Bytes(), &req); err != nil {
+			rpcError(wf, &req, rpcParseError, xerrors.Errorf("unmarshaling request: %w", err))
+			return
+		}
+		s.doHandleRequest(ctx, wf, req, *bufferedRequest, rpcError)
+
+	} else {
+		var reqs []request
+		if err := json.Unmarshal(bufferedRequest.Bytes(), &reqs); err != nil {
+			rpcError(wf, &req, rpcParseError, xerrors.Errorf("unmarshaling requests: %w", err))
+			return
+		}
```

Comment:
A single malformed request would make the whole batch fail. Is this what the JSON-RPC 2.0 spec mandates? If not, we would need to unmarshal requests individually, and record individual failures in the buffered response.